src/Storage/Processor/Zip.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+declare(strict_types=1);
+
+namespace EMS\CommonBundle\Storage\Processor;
+
+use Symfony\Component\Filesystem\Filesystem;
+
+class Zip
+{
+    /** @var Config */
+    private $config;
+
+    public function __construct(Config $config)
+    {
+        $this->config = $config;
+    }
+
+    public function generate()
+    {
+        $filesystem = new Filesystem();
+
+        $tempFile = $filesystem->tempnam(\sys_get_temp_dir(), 'emss');
+        $zip = new \ZipArchive();
```

Comment:
Use https://github.com/maennchen/ZipStream-PHP
And check $zip->addFileFromStream('goodbye.txt', $fp);